base/compiler/ssair/ir.jl

```diff
@@ -146,8 +146,22 @@ function first_insert_for_bb(code, cfg::CFG, block::Int)
     end
 end
 
-
-const NewNode = Tuple{Int, #= reverse affinity =#Bool, Any, Any, #=LineNumber=#Int}
+struct NewNode
+    # Insertion position (interpretation depends on which array this is in)
+    pos::Int
+    # Place the new instruction after this instruction (but in the same BB if this is an implicit terminator)
+    attach_after::Bool
+    # The type of the instruction to insert
+    typ::Any
+    # The node itself
+    node::Any
+    # The index into the line number table of this entry
+    line::Int
+end
+function NewNode(nnode::NewNode; pos=nnode.pos, attach_after=nnode.attach_after,
+                                 typ=nnode.typ, node=nnode.node, line=nnode.line)
```

Comment:
don't use kwargs in inference. don't make kwargs of types. especially don't make kwargs of `node`s